components/camel-oaipmh/src/main/java/org/apache/camel/oaipmh/component/OAIPMHProducer.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.oaipmh.component;
+
+import java.net.URI;
+import java.util.List;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.oaipmh.handler.Harvester;
+import org.apache.camel.oaipmh.handler.ProducerResponseHandler;
+import org.apache.camel.support.DefaultProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The OAIPMH producer.
+ */
+public class OAIPMHProducer extends DefaultProducer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OAIPMHProducer.class);
+    private OAIPMHEndpoint endpoint;
+
+    public OAIPMHProducer(OAIPMHEndpoint endpoint) {
+        super(endpoint);
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        Harvester harvester = new Harvester(
+                new ProducerResponseHandler(),
+                endpoint.getUrl(),
+                endpoint.getVerb(),
+                endpoint.getMetadataPrefix(),
+                endpoint.getUntil(),
+                endpoint.getFrom(),
+                endpoint.getSet(),
+                endpoint.getIdentitier());
+        overrideHarvesterConfigs(exchange.getIn(), harvester);
+        if (endpoint.isIgnoreSSLWarnings()) {
+            harvester.getHttpClient().setIgnoreSSLWarnings(true);
+        }
+        List<String> synHarvest = harvester.synHarvest(endpoint.isOnlyFirst());
+        exchange.getMessage().setBody(synHarvest);
+        if (endpoint.isOnlyFirst() && harvester.getResumptionToken() != null) {
+            exchange.getMessage().setHeader("CamelOaimphResumptionToken", harvester.getResumptionToken());
+        } else {
+            exchange.getMessage().removeHeader("CamelOaimphResumptionToken");
+        }
+    }
+
+    private void overrideHarvesterConfigs(Message msg, Harvester harvester) {
+        String header = msg.getHeader("CamelOaimphUrl", String.class);
+        if (header != null) {
+            harvester.setBaseURI(URI.create(header));
+        }
+
+        header = msg.getHeader("CamelOaimphEndpointUrl", String.class);
+        if (header != null) {
```

Comment:
I think these kind of validations, could be simplified into a function that takes lambda as argument, example:
```
private <T> void checkAndSetConfigs(final Message message, final String key, final Consumer<T> fn, final Class<T> type) {
        final T header = message.getHeader(key, type);
        if (!ObjectHelper.isEmpty(header)) {
            fn.accept(header);
        }
    }
``` 
to call it:
```
checkAndSetConfigs(msg, "CamelOaimphVerb", harvester::setVerb, String.class);
```
For the URI, you may need to add the a wrapper from URI to String somewhere that creates `URI.create(header)`